src/class-metadata.php

```diff
@@ -0,0 +1,632 @@
+<?php
+/**
+ * Metadata class
+ *
+ * @package Parsely
+ * @since 3.3.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+use WP_User;
+use WP_Post;
+
+/**
+ * Generates and inserts metadata readable by the Parse.ly Crawler.
+ *
+ * @since 1.0.0
+ * @since 3.3.0 Moved from class-parsely to separate file
+ */
+class Metadata {
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Creates parsely metadata object from post metadata.
+	 *
+	 * @param array<string, mixed> $parsely_options parsely_options array.
+	 * @param WP_Post              $post object.
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function construct_metadata( array $parsely_options, WP_Post $post ): array {
```

Comment:
Would there be an advantage to splitting this up into smaller methods now, rather than iterate on this later? 177 lines seems a lot to add as a "new" method, even if it is just moving it from a different class.

Could also look at making the second arg optional now, since we know that the only bit of code calling this new method is ourselves, and it's not needed for many of the conditions/contexts.

In fact, we could think about whether the $parsely_options arg is needed, since that should be consistent between requests (and for tests we update the DB anyway).